packages/public/server/src/module/Api/src/Controller/ApiController.php

```diff
@@ -69,6 +81,100 @@ public function aliasAction()
+            return $this->response;
+        }
+
+        $authorizationResponse = $this->assertAuthorization();
+        if ($authorizationResponse) {
+            return $authorizationResponse;
+        }
+
+        try {
+            $data = Json::decode(
+                $this->getRequest()->getContent(),
+                Json::TYPE_ARRAY
+            );
+        } catch (JsonRuntimeException $exception) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return $this->response;
+        }
+
+        $isDataValid =
+            is_array($data) &&
+            isset($data['title']) &&
+            is_string($data['title']) &&
+            isset($data['content']) &&
+            is_string($data['content']) &&
+            isset($data['userId']) &&
+            is_numeric($data['userId']) &&
+            isset($data['objectId']) &&
+            is_numeric($data['objectId']);
+
+        if (!$isDataValid) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return $this->response;
+        }
+
+        try {
+            $user = $this->getUserManager()->getUser($data['userId']);
+            $this->getServiceLocator()
+                ->get('Zend\Authentication\AuthenticationService')
+                ->setIdentity($user);
+        } catch (UserNotFoundException $exception) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return $this->response;
+        }
+
+        try {
+            $uuid = $this->getUuidManager()->getUuid(
+                $data['objectId'],
+                false,
+                false
+            );
+        } catch (NotFoundException $exception) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return $this->response;
+        }
+
+        if (
+            $uuid instanceof EntityInterface ||
+            $uuid instanceof PageRepositoryInterface ||
+            $uuid instanceof CommentInterface ||
+            $uuid instanceof TaxonomyTermInterface
+        ) {
+            $instance = $uuid->getInstance();
+        } else {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return $this->response;
+        }
+
+        if ($uuid instanceof CommentInterface) {
+            // TODO
+        } else {
+            /** @var DiscussionForm $form $form */
+            $form = $this->getServiceLocator()->get(DiscussionForm::class);
+            $form->setData([
+                'object' => $uuid,
+                'author' => $user,
+                'instance' => $instance,
+                'title' => $data['title'],
+                'content' => $data['content'],
+                'csrf' => CsrfTokenContainer::getToken(),
+                'subscription' => [
+                    'subscribe' => true,
+                    'mailman' => true,
+                ],
+            ]);
```

Comment:
Note to me: `$form->setData()` can only be called once (a new call overrides the old data).